cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -0,0 +1,160 @@
+
+            object FINALISING_TRANSACTION : Step("Obtaining notary signature and recording transaction.") {
+                override fun childProgressTracker() = FinalityFlow.tracker()
+            }
+
+            fun tracker() = ProgressTracker(
+                    CREATING_COMMAND,
+                    GENERATING_TRANSACTION,
+                    VERIFYING_TRANSACTION,
+                    SIGNING_TRANSACTION,
+                    INITIALIZE_OTHER_PARTY_FLOW,
+                    GATHERING_SIGS,
+                    PAYBACK_COMMUNICATION,
+                    FINALISING_TRANSACTION
+            )
+        }
+
+        override val progressTracker = tracker()
+
+        /**
+         * The flow logic is encapsulated within the call() method.
+         */
+        @Suspendable
+        override fun call(): SignedTransaction {
+            progressTracker.currentStep = CREATING_COMMAND
+            //val signedTransaction = serviceHub.validatedTransactions.getTransaction(loanTxId)
+            // val signedTransaction = transactions.find { it.id == transactionHash } ?: throw IllegalArgumentException("Unknown transaction hash.")
+
+            /* val criteria = QueryCriteria.VaultQueryCriteria(
+                    contractStateTypes = setOf(IOUState::class.java),
+                    status = Vault.StateStatus.UNCONSUMED,
+                    exactParticipants = listOf(ourIdentity, otherParty)
+            ) // */
+
+            val uuid: UUID = UUID.fromString(inputLinearId)
+            val queryCriteria: QueryCriteria = LinearStateQueryCriteria().withUuid(Collections.singletonList(uuid))
+
+            val results = serviceHub.vaultService.queryBy(IOUState::class.java, queryCriteria)
+            // val ourStateRef = StateRef(SecureHash.parse(loanTxId), 0)
+            // val (state, ref) = serviceHub.toStateAndRef<ContractState>(ourStateRef)
+            val inputStateAndRef = serviceHub.toStateAndRef<ContractState>(results.states[0].ref)
+            val inputIOUState = inputStateAndRef.state.data as IOUState
+
+            progressTracker.currentStep = GENERATING_TRANSACTION
+            // Generate an unsigned transaction.
+            val txCommand = Command(IOUContract.Commands.Destroy(), inputIOUState.participants.map { it.owningKey })
+
+            // Obtain a reference to the notary that was in use for input state
+            val notary = results.states[0].state.notary
+
+            // build tx that will consume previous output
+            val txBuilder = TransactionBuilder(notary)
+                    .addInputState(inputStateAndRef)
+                    .addCommand(txCommand)
+
+            progressTracker.currentStep = VERIFYING_TRANSACTION
+            // Verify that the transaction is valid.
+            txBuilder.verify(serviceHub)
+
+            progressTracker.currentStep = SIGNING_TRANSACTION
+            // Sign the transaction (must be by a borrower)
```

Comment:
Don't make it a comment, make it a strict check.

Reply:
Or make your code agnostic as to whether which party started it, since they both need to sign anyway, according to you.